library/core/src/ffi/c_str.rs

```diff
@@ -196,15 +196,16 @@ impl CStr {
     /// allows inspection and interoperation of non-owned C strings. The total
     /// size of the raw C string must be smaller than `isize::MAX` **bytes**
     /// in memory due to calling the `slice::from_raw_parts` function.
-    /// This method is unsafe for a number of reasons:
     ///
-    /// * There is no guarantee to the validity of `ptr`.
-    /// * The returned lifetime is not guaranteed to be the actual lifetime of
-    ///   `ptr`.
-    /// * There is no guarantee that the memory pointed to by `ptr` contains a
-    ///   valid nul terminator byte at the end of the string.
-    /// * It is not guaranteed that the memory pointed by `ptr` won't change
-    ///   before the `CStr` has been destroyed.
+    /// # Safety
+    ///
+    /// * `ptr` must be valid for reads of `u8`.
+    /// * The caller must ensure that the returned lifetime represents the actual
+    ///   lifetime of `ptr`.
```

Comment:
It is not entirely clear what this means, either.

I think it would probably best to use the same terminology as in https://doc.rust-lang.org/nightly/std/slice/fn.from_raw_parts.html, since this is a very similar kind of operation.

Reply:
I don't think this is resolved. "the actual lifetime of `ptr`" is not a well-defined concept I think, and the other docs I mentioned don't use terminology like that.

Instead, the key phrase is:
"The memory referenced by the returned `CStr` not be mutated for the duration of lifetime `'a`"

That also makes your last bullet redundant.